### PR TITLE
docs(changelog): added note about #13420 to the CHANGELOG, which affe…

### PR DESCRIPTION
**Pull request by vandres**
…cts incorrect usages of trackBy

### CODE REVIEW

**Comment by vandres** [comment]
I signed it!

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,7 +18,10 @@ Note: 4.0.0-beta.1 release also contains all the changes present in the 2.4.0 an
 * **core:** improve error message when component factory cannot be found ([#13541](https://github.com/angular/angular/issues/13541)) ([b9e979e](https://github.com/angular/angular/commit/b9e979e)), closes [#12678](https://github.com/angular/angular/issues/12678)
 * **router:** should reset location if a navigation by location is successful ([#13545](https://github.com/angular/angular/issues/13545)) ([a38f14b](https://github.com/angular/angular/commit/a38f14b)), closes [#13491](https://github.com/angular/angular/issues/13491)
 
+### Note ###
 
+With [#13420](https://github.com/angular/angular/pull/13420) incorrect usages of `trackBy` will lead to an error which may break your application. The only valid usage of `trackBy` is providing a function, as stated
```
Comment (vicb) [inline CHANGELOG.md]: Don't agree with the wording: it will not break the app as the app is already broken.
We now enforce that a function is given.

Comment (DzmitryShylovich) [inline CHANGELOG.md]: It will break 3rd party libs which rely on the fast that null == identity comparison https://github.com/angular/angular/issues/13641

**Comment by vicb** [comment]
blocked: this seems to cause issue, might get reverted. @mhevery ?

**Pull request closed without merging**